packages/core-web/src/Wallet.ts

```diff
@@ -0,0 +1,177 @@
+      )
+    }
+
+    if (this._federationId) {
+      throw new Error(
+        `Wallet ${this.id} is already part of federation ${this._federationId}`,
+      )
+    }
+
+    try {
+      logger.debug('called joinFederation with invite code:', inviteCode)
+      const res = await this._client.joinFederation(
+        inviteCode,
+        this._clientName,
+      )
+      this._federationId = (
+        await this._client.parseInviteCode(inviteCode)
+      ).federation_id
+      this._isOpen = true
+      this._resolveOpen()
+      logger.info(
+        `Wallet ${this.id} successfully joined federation ${this._federationId}`,
+      )
+      return true
+    } catch (e) {
+      logger.error(`Error joining federation for wallet ${this.id}:`, e)
+      this._isOpen = false
+      this._federationId = undefined
+      return false
+    }
+  }
+
+  /**
+   * Opens the existing wallet.
+   *
+   * It resolves when the wallet has successfully opened.
+   *
+   * @returns {Promise<void>} A promise that resolves when the wallet is open.
+   */
+  async open(federationId: string): Promise<boolean> {
+    if (this._isOpen) {
+      throw new Error('The FedimintWallet is already open.')
+    }
+
+    try {
+      // Ensure the RPC client is initialized before making the call
+      await this._client.initialize()
+
+      // Then open the client
+      await this._client.openClient(this._clientName)
+      this._federationId = federationId
+      this._isOpen = true
+      this._resolveOpen()
+      logger.info(
+        `Wallet ${this.id} opened successfully with federation ${this._federationId}`,
+      )
+      return true
+    } catch (e) {
+      logger.error(`Error opening wallet ${this.id}:`, e)
+      return false
+    }
+  }
+
+  /**
+   * Cleans up the wallet resources.
+   *
+   * This method closes the client and removes the wallet
+   * from the WalletManager. It should be called when the wallet is no longer needed.
+   *
+   * @returns {Promise<void>} A promise that resolves when the cleanup is complete.
+   */
+  async cleanup(): Promise<void> {
+    try {
+      if (this._isOpen) {
+        await this._client.closeClient(this._clientName)
+      }
+      // WalletManager.getInstance().removeWallet(this.id)
+      this._isOpen = false
+      this._openPromise = undefined
+      logger.info(`Wallet ${this.id} cleaned up`)
+    } catch (error) {
+      logger.error(`Error cleaning up wallet ${this.id}:`, error)
+    }
+  }
+
+  /**
+   * Checks if the wallet is currently open.
+   * @returns {boolean} True if the wallet is open, false otherwise.
+   */
+  isOpen(): boolean {
+    return this._isOpen
+  }
+
+  // Debug/Testing methods
+  _getActiveSubscriptionIds(): number[] {
+    return this._client._getActiveSubscriptionIds()
+  }
+
+  _getActiveSubscriptionCount(): number {
+    return this._client._getActiveSubscriptionCount()
+  }
+
+  _getRequestCounter(): number {
+    return this._client._getRequestCounter()
+  }
```

Comment:
remove this, should be on wallet directory. actually do tests get a separate wallet director or is it shared?